Pull request to review changes - all tests to use local web driver

skondkar opened the pull request.

skondkar committed:
New branch for all MS4 tests.

skondkar committed:
testng.xml to only include MS4 tests.

skondkar committed:
Removed HawkularRedHatAccess.java from utils and added HawkularRedHatAccessPage.java to page package. Modified HawkularRHASearchTest.java against slow RH Access connection.

skondkar committed:
Refactored UI tests to use selenium local web driver. Modified failing HawkualrLoginTest.

FilipB commented:
A few general comments:
- There is no javadoc at all - at least a few lines at class level and for more complex methods would be really good.
- There is no logging - logging is very useful when debugging unstable tests (works locally but fails on CI server) - see e.g. http://www.javacodegeeks.com/2011/01/10-tips-proper-application-logging.html
- Methods in page objects could return another page object so you can call more operations at one line without a need to create new instances e.g - page1.navigateToPage2().openCase().insertValues().submit();
- Another good improvement could be check for successful and error message boxes (if there are any in hawkular) - e.g. after creating new case, check that successful msg box is visible. For negative cases, check that err msg box is visible. If not, fail the test.

jkandasa commented on src/main/java/org/qe/hawkular/page/HawkularCreateCasePage.java:
```diff
@@ -0,0 +1,87 @@
+	By selectSeverity = HawkularCreateCaseConstants.selectSeverity;
+	By clickSubmit = HawkularCreateCaseConstants.clickSubmit;
+
+
```
@skondkar I feel, creating local reference is not require, you can call directly from `HawkularCreateCaseConstants`

FilipB replied:
I believe this was done to shorten subsequent code length. Another option could be static import although it is not the best practice.

jkandasa commented on src/main/java/org/qe/hawkular/page/HawkularCreateCasePage.java:
```diff
@@ -0,0 +1,87 @@
+	public void switchFrameFocus() {
+		driver.switchTo().frame("openCaseWindow");		
+	} 
+
```
@skondkar If `switchFrameFocus` used in many part of ui automation you can put it on some common location, later if you wan to change anything on `switchFrameFocus` you can do in on one place.

skondkar committed:
Added javadoc comments at class level.

skondkar commented:
Thanks for the review.!
- Added javadoc at class level, will be adding more at complex methods- Ticket to track- #23
- Added a ticket to track for logging - #22
- We will make sure to have check for successful and error message boxes.
- The testng was earlier having only MS4 tests to be able for Matt to have Polarion export only for Ms4 tests. Now the testng.xml includes all the tests since MS1.

jkandasa commented:
@skondkar Thank you Sunil!
